frontend/src/actionCreators/parcelsActionCreator.ts

```diff
@@ -81,38 +81,39 @@ export const fetchPropertyDetail = (
     : dispatch(fetchBuildingDetail({ id }, position));
 };
 
-export const createParcel = (parcel: IParcel) => (dispatch: Function) => {
+export const createParcel = (parcel: IParcel) => async (dispatch: Function) => {
   dispatch(request(actionTypes.ADD_PARCEL));
   dispatch(showLoading());
-  return CustomAxios()
-    .post(ENVIRONMENT.apiUrl + API.PARCEL_ROOT, parcel)
-    .then((response: AxiosResponse) => {
-      dispatch(success(actionTypes.ADD_PARCEL, response.status));
-      dispatch(fetchParcelDetail(response.data));
-      dispatch(hideLoading());
-    })
-    .catch((axiosError: AxiosError) => {
-      dispatch(error(actionTypes.ADD_PARCEL, axiosError?.response?.status, axiosError));
-      throw Error(axiosError.response?.data.details);
-    })
-    .finally(() => dispatch(hideLoading()));
+  try {
+    const { data, status } = await CustomAxios().post(ENVIRONMENT.apiUrl + API.PARCEL_ROOT, parcel);
+    dispatch(success(actionTypes.ADD_PARCEL, status));
+    dispatch(fetchParcelDetail(data));
+    dispatch(hideLoading());
+    return data;
+  } catch (axiosError) {
+    dispatch(error(actionTypes.ADD_PARCEL, axiosError?.response?.status, axiosError));
+    dispatch(hideLoading());
+    throw Error(axiosError.response?.data.details);
+  }
 };
 
-export const updateParcel = (parcel: IParcel) => (dispatch: Function) => {
+export const updateParcel = (parcel: IParcel) => async (dispatch: Function) => {
   dispatch(request(actionTypes.UPDATE_PARCEL));
```

Comment:
It would be great to abstract a lot of this out to common functions, we're seeing repetition far too much.  Makes it hard to maintain and slows development.  Don't make changes in this PR though

Reply:
one solution is handleAxiosResponse in utils.ts